src/language-html/embed.js

```diff
@@ -381,16 +382,18 @@ function embed(path, print, textToDoc, options) {
         options
       );
       if (embeddedAttributeValueDoc) {
-        return [
-          node.rawName,
-          '="',
-          group(
-            mapDoc(embeddedAttributeValueDoc, (doc) =>
-              typeof doc === "string" ? doc.replace(/"/g, "&quot;") : doc
-            )
-          ),
-          '"',
-        ];
+        const favoriteQuote = options.singleQuote ? "'" : '"';
+        let quote = favoriteQuote;
+        const content = mapDoc(embeddedAttributeValueDoc, (doc) => {
+          if (typeof doc !== "string") {
+            return doc;
+          }
+
+          const preferred = getPreferredQuote(doc, favoriteQuote);
+          quote = preferred.quote;
```

Comment:
I think this is not right, the `embeddedAttributeValueDoc` can have different "preferred" quote, it should not decide by the last one.

Reply:
**Prettier pr-11559**
[Playground link](https://deploy-preview-11559--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmpZkBuAtNALwDkYeOOAFMSAIw8AaMgDI8WCAGcA3Dzyjx0oT3mSpAIx4BKZoVQB6LCSj7MuAnEIgBICFhgBLaBOSg8AJzcQA7gAV3CZxR2LzwAT2drNTc8MABrOBgAZTwMOAAZeyg4ZAAzdgk4SOi4hMSsGMyAc2QYNwBXQpA4DDU4ABM29rS8KEq6vEq4ADEINwwCB17kEDw6mAgrEAALGAwcAHUl+3gJcrA4RIDt+xpt0OmwCQiQTIK3GB9oyvHc-MaAKwkAD0SqnDgAIp1CDwV44ArWcpuO7TGgNRZYNyZGDrextGBLZAADgADJDPAV1tEsNNEXA7jRstYAI7A+CPWyBGYSOhZdrtRZuOC0+xcx4DF5IPLgxoFDD2Gr1UV-QF07JIWoNawwPBqVHozFIABMyui9hwVQAwhAMIKmhIAKyLOoFAAqqsCwohIDhcAAklBOrBEmAkXYAIKexIwUL-MEFAC+EaAA)
<!-- prettier-ignore -->
```sh
--parser vue
--single-quote
```

**Input:**
<!-- prettier-ignore -->
```vue
<template>
  <p v-on='call("1", &apos;"a&apos;, "&apos;b")'></p>
</template>
```

**Output:**
<!-- prettier-ignore -->
```vue
<template>
  <p v-on='call('1', '&quot;a', '\'b')'></p>
</template>

```